### PR TITLE
Optimized for autocomplete+ snippets

### PR DESCRIPTION
**Pull request by imanghafoori1**
https://github.com/atom-community/autocomplete-snippets
- note that this branch heavily depends on the package called autocomplete-snippets for it's effectiveness.
- every snippet is preceded with "em" for Ember or ds for DS. the reason for that is to Namespace the snippets so they do not collide if another package defines snippets for model,view,.... of backbone or angular.
- no need for --help or shorthands, thanks to the autocomplete-snippets package.
  try the snippets below:
  emView
  emArrayController
  dsModel
  emComputed
  ,...

### CODE REVIEW

**Comment by seifsallam** [comment]
Awesome work, but there are still some comments regarding the changes.

I think prefixes needs more work. Some has short hand, some has underscore, some has dashes, some has both, and some had dots

For starters, lets use full names for everything. So using `Ember` instead of `Em`, and `classNames instead of `cleassN`, ...etc and so on for the other cases.

For separating the words, I don't think it's a good idea to use underscores, it's not really a JavaScript convention, so i'm leaning more toward matching syntax with the function itself. That means using CamelCase to call `FixureAdapter` and so on for the rest of prefixes. 

For case that require use of dot, lets use dot. like `computed.or`

For using DS, and Ember. I don't agree that we have to namespace it. Unless there is a conflict locally, we should use direct function names. For Angular and other users, I think that would be a rare case to stumble upon. And in that instance, they can disable this snippet.

For es6, it should be `import-ember` instead `es6`, because it's a snippet for import. `import` in this case would be custom. Also there is no need to have export default snippet.

So as a rule, lets just match the prefix with the real syntax. And if we can avoid using DS or Ember, that would be even better. It will match people's expectations

Also, there should be maximum of one line spacing between text lines in all files.

**Comment by imanghafoori1** [comment]
On the last commits I changed all of them to _ underscores.
And prefixes are now all uniform and consistent,

em_view_...
em_route_...
...

**Comment by seifsallam** [comment]
Awesome, but still. As I said above. It's better to match the prefix to the real sytanx of what people would write.

**Comment by imanghafoori1** [comment]
I too think that em_ is pretty nasty but as a global package, huuum we have to have something special.
For my personal use I remove it without mercy.
but route, view, controller is common in backbone too.
Any way, if you think it is better to remove them, feel free to edit them on your own.

**Comment by imanghafoori1** [comment]
I first tried with the dot syntax em.view.somthing but the auto completion  did not worked with dot properly.
so I chose the underscores.

**Comment by imanghafoori1** [comment]
For webstorm and phpstorm I do change it to dot syntax since it works there,

**Comment by imanghafoori1** [comment]
sorry I forgot to upload the local changes of route.cson file.

**Comment by imanghafoori1** [comment]
I think it is safer to follow the community demand of namespace and provide an optional wrapper.cson file to translate "computed_or" to "em_computed_or" for the users to use. and put the instructions of add wrapper.cson in the README file.
since we live a in crowded world, nowadays, it is likely to have these conflicts around.
And we are forced to use namespaces. ;)

**Comment by seifsallam** [comment]
I understand your point of view, but I still think the number of people who are using Ember with other MVC is pretty small, not to change the entire naming for that. If i'm wrong and the number is higher or got higher, people will complain, and then we can add namespacing. But for the majority now. Ember should be first class citizen and should be treated as such

**Comment by imanghafoori1** [comment]
Ok, I upload the em_ removed version on a new branch. in a few minutes.
and Note that I hear Ember experts tell that views are used rarely and apps.

**Comment by seifsallam** [comment]
Please don't open another branch

**Comment by seifsallam** [comment]
Let's just keep things consistent here

**Comment by imanghafoori1** [comment]
They are ready, I think.

Just forgive me about the extra linebreakes.
 I remove them in future commits.
They now cause no overload.

File: snippets/view.cson [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    'body':
+      """
+        parentViewDidChange: ->
+          ${1:#logic...}
+      """
+
+
+
+  'Ember willClearRender Hook':
+    'prefix': 'view_willClearR'
```
Comment (seifsallam) [inline snippets/view.cson]: Can you fix this

File: snippets/view.cson [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+      """
+
+
+
+
+  'Ember.Select.extend(...':
+    'prefix': 'view_Select'
+    'body':
+      """
+        Ember.Select.extend
```
Comment (seifsallam) [inline snippets/view.cson]: Not sure, but Isn't Ember.Select deprecated?

File: snippets/router.cson [hunk 3]
```diff
@@ -0,0 +1,17 @@
+'.source.coffee':
+  'Nested resource':
+    'prefix': '@resource-v'
```
Comment (seifsallam) [inline snippets/router.cson]: Prefix is inconsitent

File: snippets/model.cson [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    'prefix': 'model_attr'
+    'body': "${1}: DS.attr()"
+
+
+
+
+
+
+  'Verbose DS.hasMany(...':
+    'prefix': 'model_hasMany-v'
```
Comment (seifsallam) [inline snippets/model.cson]: Inconsistent prefix

File: snippets/adapters.cson [hunk 5]
```diff
@@ -0,0 +1,40 @@
+'.source.coffee':
+
+  'Ember FixtureAdapter':
+    'prefix': 'adapter_fixture'
```
Comment (seifsallam) [inline snippets/adapters.cson]: `fixture_adapter`

File: snippets/view.cson (hunk 2 above)
Comment (imanghafoori1) [inline snippets/view.cson]: huuum, I don't know exactly, I just read about it on the official docs.

File: snippets/model.cson (hunk 4 above)
Comment (imanghafoori1) [inline snippets/model.cson]: it give the user an option to choose the simple or complex version.
"-v" is usually used in command line (terminal) as a flag, so I use it here.

File: snippets/adapters.cson [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    'body':
+      """
+        _export = DS.FixtureAdapter.extend()
+      """
+
+
+  'Ember RESTAdapter':
+    'prefix': 'adapter_rest'
```
Comment (seifsallam) [inline snippets/adapters.cson]: `rest_adapter`

File: snippets/adapters.cson [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+          {
+            id: 2
+            ${3}: "${4}"
+          }
+        ]
+      """
+
+
+  'Ember ActiveModelAdapter':
+    'prefix': 'adapter_activeModel'
```
Comment (seifsallam) [inline snippets/adapters.cson]: activeModel-adapter

File: snippets/router.cson (hunk 3 above)
Comment (imanghafoori1) [inline snippets/router.cson]: it is for cases they are tried to type what they often do.
It set a trap ;)
they start to type @reso... they are offered with snippet above.

Comment (seifsallam) [inline snippets/router.cson]: ic, but there is also the `-v` part

**Comment by imanghafoori1** [comment]
I tried to group all adapters under the adapter keyword,
when the user need an adapter he simply type adap... and him will be offered with 3 types and chooses one.
ok, I also provide the other prefix to.
Good idea.

**Comment by imanghafoori1** [comment]
Again, Happy to have you as a teammate.
your comments drove the project to reach a very good point.
I continue to work on this with the accepted principals.

**Pull request closed without merging**